src/main/java/com/example/demo/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,52 @@
+package com.example.demo.controller;
+
+import com.example.demo.dto.cart.item.CartItemRequestDto;
+import com.example.demo.dto.cart.item.CartItemResponseDto;
+import com.example.demo.dto.shopping.cart.ShoppingCartDto;
+import com.example.demo.service.cart.item.CartItemService;
+import com.example.demo.service.shopping.cart.ShoppingCartService;
+import com.example.demo.service.user.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cart")
+@RequiredArgsConstructor
+public class ShoppingCartController {
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
+    private final CartItemService cartItemService;
+
+    @GetMapping
+    public ShoppingCartDto getShoppingCart(Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        Long currentUser = userService.getUserIdByEmail(userDetails.getUsername());
+        return shoppingCartService.getByUserId(currentUser);
+    }
+
+    @PostMapping
+    public CartItemResponseDto addBook(@RequestBody CartItemRequestDto cartItemRequestDto,
+                                       Authentication authentication) {
+        return cartItemService.saveItemToShoppingCart(cartItemRequestDto,
+            getShoppingCart(authentication));
```

Comment:
In Spring Boot, it's generally not recommended for controller methods to directly call each other. Controller methods are typically designed to handle incoming HTTP requests and delegate processing to services or other components. Directly calling one controller method from another can lead to tight coupling between them, making the code harder to understand, test, and maintain.

Instead, it's better to extract common functionality into separate service classes and have the controller methods delegate to these services. This promotes separation of concerns and allows for better code organization and reusability. Additionally, services can be easily tested in isolation, which is crucial for maintaining code quality.